### PR TITLE
feat: integrate DAOStack. Add the corresponding checker and test.

### PR DESCRIPTION
**Pull request by Soptq**
This PR integrates DAOStack. User's membership is verified by checking his DAOToken or Reputation balance.

Reference:
https://github.com/daostack/arc/blob/master/contracts/controller/Controller.sol#L38-L39

### CODE REVIEW

File: contracts/membershipCheckers/DAOStackMembershipChecker.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        public
+        view
+        override
+        returns (bool)
+    {
+        require(daoAddress != address(0), "AutID: daoAddress empty");
+        require(member != address(0), "AutID: member empty");
+
+        // implement your membership checker logic here
+        // return reasonable result ;)
```
Comment (migrenaa) [inline contracts/membershipCheckers/DAOStackMembershipChecker.sol]: please delete these comments.

File: contracts/membershipCheckers/DAOStackMembershipChecker.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "./IMembershipChecker.sol";
+import "../daoStandards/IDAOStack.sol";
+
+/// @title NewMembershipChecker
```
Comment (migrenaa) [inline contracts/membershipCheckers/DAOStackMembershipChecker.sol]: put relevant title, notice etc, please

**Review by migrenaa — CHANGES_REQUESTED** [review]

File: contracts/membershipCheckers/DAOStackMembershipChecker.sol (hunk 2 above)
Comment (Soptq) [reply contracts/membershipCheckers/DAOStackMembershipChecker.sol]: Done ;)

**Review by migrenaa — APPROVED** [review]

File: contracts/membershipCheckers/DAOStackMembershipChecker.sol (hunk 2 above)
Comment (migrenaa) [reply contracts/membershipCheckers/DAOStackMembershipChecker.sol]: Thanks! We're talking about anything Open source and Bounties related on our discord server. Join us there :) https://discord.gg/dRmh62Yj

**Pull request merged**